orte/mca/oob/tcp/oob_tcp_listener.c

```diff
@@ -739,8 +742,31 @@ static void* listen_thread(opal_object_t *obj)
                 pending_connection->fd = accept(sd,
                                                 (struct sockaddr*)&(pending_connection->addr),
                                                 &addrlen);
+
+                /* if we are on a privileged port, we only accept connections
+                 * from other privileged sockets. A privileged port is one
+                 * whose port is less than 1024 on Linux, so we'll check for that. */
+                if (1024 >= listener->port) {
+                    uint16_t inport;
+                    if (listener->tcp6) {
+                        inport = ntohs(((struct sockaddr_in6*)&pending_connection->addr)->sin6_port);
+                    } else {
+                        inport = ntohs(((struct sockaddr_in*)&pending_connection->addr)->sin_port);
+                    }
+                    if (1024 < inport) {
+                        /* someone tried to cross-connect privileges,
+                         * say something */
+                        orte_show_help("help-oob-tcp.txt",
+                                       "privilege failure",
+                                       true, opal_process_info.nodename,
+                                       listener->port, inport);
+                        OBJ_RELEASE(pending_connection);  // closes the incoming fd
```

Comment:
Do you intentionally __not__ close ```pending_connection->fd``` ?
Or should the destructor (to be created) do that for you ?